libs/extensions/angular/localization/src/amount/amount-service-formatter.ts

```diff
@@ -1,46 +1,48 @@
 import { formatNumber } from '@angular/common';
-
+import { KirbyExtensionsLocalizationToken } from '../di-tokens';
 import { Amount } from './amount.model';
 
 export function formatAmount(
   amount: Amount,
   locale: string,
-  nativeCurrency: string,
+  config: KirbyExtensionsLocalizationToken,
   amountServiceConfiguration?: AmountServiceConfiguration
 ) {
-  const config = deriveConfiguration(amountServiceConfiguration);
+  const amountConfig = deriveConfiguration(amountServiceConfiguration);
 
-  let formattedAmount = formatNumber(amount && amount.amount, locale, config.digitsInfo);
+  let formattedAmount = formatNumber(amount && amount.amount, locale, amountConfig.digitsInfo);
 
-  if (config.stripSign) {
+  if (amountConfig.stripSign) {
     formattedAmount = formattedAmount.replace('-', '').trim();
   }
 
-  const currencyCodeToAppend = deriveCurrencyCode(config, amount, nativeCurrency);
+  const currencyCodeToAppend = deriveCurrencyCode(amountConfig, amount, config);
 
   if (!currencyCodeToAppend) {
     return formattedAmount;
   }
-  if (config.currencyCodePosition === 'postfix') {
+  if (amountConfig.currencyCodePosition === 'postfix') {
     return formattedAmount + ' ' + currencyCodeToAppend;
   } else {
     return currencyCodeToAppend + ' ' + formattedAmount;
   }
 }
 
 export function deriveCurrencyCode(
-  config: AmountServiceConfiguration,
+  amountConfig: AmountServiceConfiguration,
   amount: Amount,
-  nativeCurrency: string
+  config: KirbyExtensionsLocalizationToken
 ) {
   let currencyCodeToAppend;
 
-  if (config.showCurrencyCode) {
-    if (config.showCurrencyCode === 'alwaysShowCurrency') {
-      currencyCodeToAppend = amount && amount.currencyCode;
-    } else if (config.showCurrencyCode === 'showForeignCurrency') {
+  if (amountConfig.showCurrencyCode) {
+    if (amountConfig.showCurrencyCode === 'alwaysShowCurrency') {
+      currencyCodeToAppend = amount.currencyCode;
+    } else if (amountConfig.showCurrencyCode === 'showForeignCurrency') {
       currencyCodeToAppend =
-        amount && amount.currencyCode !== nativeCurrency ? amount.currencyCode : '';
+        amount.currencyCode !== config.nativeCurrency ? amount.currencyCode : '';
+    } else if (amountConfig.showCurrencyCode === 'showCurrencySymbol') {
+      currencyCodeToAppend = config.currencyMappings?.[amount.currencyCode] || '';
```

Comment:
Burde vi ikke lade den defaulte til `currencyCode`? Den mest almindelige usecase vil være at tilføje speciel håndtering for DKK, uden at man skal forholde sig til ALLE andre currencies.

```suggestion
      currencyCodeToAppend = config.currencyMappings?.[amount.currencyCode] || amount.currencyCode;
```